resources/config/snemo/demonstrator/simulation/primary_events/1.3/_tools/_prepare_csv.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+# to be executed from the base config directory where manager.conf is stored
+
+which bxextract_table_of_objects 1> /dev/null
+if [ $? -ne 0 ]; then
+    echo >&2 "ERROR: No 'bxextract_table_of_objects' utility is set!"
+    exit 1
+fi
+#
+csv_target="variants/models/peg_generators.csv"
+csv_target_bak="variants/models/_peg_generators.csv.bak"
+
+echo >&2 "INFO: csv_target='${csv_target}'"
+
+if [ -f "${csv_target}" ]; then
+    # Backup:
+    cp -f ${csv_target} ${csv_target_bak}
+fi
+
+rm -f ${csv_target}
+
+# Production:
+
+opwd=$(pwd)
+resources_dir=../../../../../..
+cd ${resources_dir}
+resources_dir=$(pwd)
+echo >&2 "INFO: resources_dir='${resources_dir}'"
+cd ${opwd}
+
+common_pe_dir_1_1=${resources_dir}/config/common/simulation/primary_events/1.1/
+common_pe_dir_1_2=${resources_dir}/config/common/simulation/primary_events/1.2/
+pe_dir_1_2=${opwd}/../1.2/
+pe_dir_1_3=${opwd}/../1.3/
+
+bxextract_table_of_objects -i ${common_pe_dir_1_1}/generators/dbd.def \
+			   >> ${csv_target}
+
+bxextract_table_of_objects -i ${common_pe_dir_1_1}/generators/background.def --group "Background" -G \
+			   >> ${csv_target}
+
+bxextract_table_of_objects -i ${common_pe_dir_1_1}/generators/calibration.def --group "Calibration" -G \
+			   >> ${csv_target}
+
+bxextract_table_of_objects -i ${common_pe_dir_1_2}/generators/dbd_more.def \
+			   >> ${csv_target}
+
+bxextract_table_of_objects -i ${pe_dir_1_2}/generators/misc.def --group "Miscellaneous" -G \
+			   >> ${csv_target}
+
+bxextract_table_of_objects -i ${pe_dir_1_2}/generators/versatile.def --group "User" -G \
+			   >> ${csv_target}
+
+bxextract_table_of_objects -i ${pe_dir_1_3}/generators/versatile_more.def --group "User" -G \
+			   >> ${csv_target}
+
+echo >&2 "INFO: File '${csv_target}' has been generated!"
+
+exit 0
+
+# end
```

Comment:
Document the purpose of this file in the relevant README, including when and how it is supposed to be run.

Reply:
This script is run by the developer of the new config in order to generate a file that lists all available
generators. The resulting CSV file (which is loadable by static variant config) must be considered as a "static" resource file. So I consider it should not be  built by the build system. Maybe it could be but as it is needed during the development stage, I chose to freeze it at an early step.
So it is processed like any other resource files in this directory and must be added in the updated list of resource files.
I agree it will be better to explain when and how it is run while preparing such a new config. Docs, docs, docs!

Reply:
It feels like the resultant `csv` file is "compiled" from the input `.def` files. In other words, it's duplicating information that's already in the `.def` files, and is a bit like a `.o` file.

What happens if the developer _forgets_ to regenerate the `csv` file and commits changes to the `def` file?